packages/ui/src/ui/components/JobStatisticTable/JobStatisticTable.tsx

```diff
@@ -0,0 +1,180 @@
+import './Statistics.scss';
+
+const block = cn('job-statistics');
+
+const ElementsTable: any = ElementsTableRaw;
+
+interface AvgProps {
+    item: JobTreeItem;
+}
+
+function Avg({item}: AvgProps) {
+    const statistic: LeafStatistic = item.attributes.value as LeafStatistic;
+
+    if (statistic && statistic.count && statistic.sum) {
+        const result: number = statistic.sum / statistic.count;
+
+        if (result < 1) {
+            return hammer.format['Number'](result, {significantDigits: 6});
+        } else {
+            return hammer.format.Number(result);
+        }
+    }
+
+    return hammer.format.NO_VALUE;
+}
+
+interface StatisticProps {
+    item: JobTreeItem;
+    aggregation: 'avg' | 'min' | 'max' | 'sum' | 'count' | 'last';
+}
+
+export function JobStatisticTableStaticCell({item, aggregation}: StatisticProps) {
+    if (item.isLeafNode) {
+        const statistic: LeafStatistic = item.attributes.value as LeafStatistic;
+
+        if (aggregation === 'avg') {
+            return <Avg item={item} />;
+        } else {
+            return hammer.format['Number'](statistic[aggregation]);
+        }
+    }
+
+    return hammer.format.NO_VALUE;
+}
+
+interface ItemState {
+    empty: boolean;
+    collapsed: boolean;
+    visible: boolean;
+}
+
+interface MetricProps<Item extends Partial<JobTreeItem>> {
+    item: Item;
+    itemState: ItemState;
+    toggleItemState: Function;
+    renderValue: (item: Item) => React.ReactChild;
+    minWidth?: number;
+}
+
+export function ExpandedCell<Item extends Partial<JobTreeItem>>({
+    item,
+    itemState,
+    toggleItemState,
+    minWidth = undefined,
+    renderValue,
+}: MetricProps<Item>) {
+    const offsetStyle = React.useMemo(() => {
+        return {minWidth, paddingLeft: (item?.level || 0) * LEVEL_OFFSET};
+    }, [item.level, minWidth]);
+
+    const toggleItemAndTreeState = useCallback(() => {
+        if (!itemState.empty) {
+            toggleItemState();
+        }
+    }, [itemState, toggleItemState]);
+
+    if (item.isLeafNode) {
+        return (
+            <span className={block('metric')} style={offsetStyle}>
+                <Icon awesome="chart-line" className={block('metric-icon')} />
+
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    } else {
+        return (
+            <span className={block('group')} style={offsetStyle} onClick={toggleItemAndTreeState}>
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'angle-down' : 'angle-up'}
+                    className={block('group-icon-toggler')}
+                />
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'folder' : 'folder-open'}
+                    className={block('group-icon')}
+                />
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    }
+}
+
+export type TableProps<Item extends Partial<JobTreeItem>> = {
+    columns?: {
```

Comment:
templates should be described in the component, and all we need instead of `TableProps` is a list of visible columns in properties of `JobStatisitcsTable`, something like:
```ts
visibleColumns: Array<'avg' | 'min' | 'max' | 'sum' | 'count' | 'last'>
```
We don't wan't any `ElementsTable`-specific properties for `GeneralJobStatisticTable`